src/host/ledger.h

```diff
@@ -625,6 +625,35 @@ namespace asynchost
           max_chunk_threshold_size));
       }
 
+      // Recover last idx from read-only ledger directories
+      for (const auto& read_dir : read_ledger_dirs)
+      {
+        LOG_DEBUG_FMT("Recovering read-only ledger directory \"{}\"", read_dir);
+        if (!fs::is_directory(read_dir))
+        {
+          throw std::logic_error(
+            fmt::format("\"{}\" is not a directory", read_dir));
+        }
+
+        for (auto const& f : fs::directory_iterator(read_dir))
+        {
+          auto last_idx_ = get_last_idx_from_file_name(f.path().filename());
+          if (!last_idx_.has_value())
+          {
+            LOG_DEBUG_FMT(
+              "Read only ledger file \"{}\" is ignored as not committed",
```

Comment:
```suggestion
              "Read-only ledger file \"{}\" is ignored as not committed",
```

I don't feel strongly either way, but we hyphenate everywhere I can see, and I think consistency would be good.